astropy/io/fits/fitstime.py

```diff
@@ -550,20 +550,17 @@ def time_to_fits(table):
         # By default, Time objects are written in full precision, i.e. we store both
         # jd1 and jd2 (serialize_method['fits'] = 'jd1_jd2'). Formatted values for
         # Time can be stored if the user explicitly chooses to do so.
+        col_cls = MaskedColumn if col.masked else Column
         if col.info.serialize_method['fits'] == 'formatted_value':
-            newtable.replace_column(col.info.name, Column(col.value))
+            newtable.replace_column(col.info.name, col_cls(col.value))
             continue
 
         # The following is necessary to deal with multi-dimensional ``Time`` objects
         # (i.e. where Time.shape is non-trivial).
-        jd12 = np.array([col.jd1, col.jd2])
+        jd12 = np.stack([col.jd1, col.jd2], axis=-1)
```

Comment:
This avoids the `np.roll` right after, but is not strictly required right now. But I will need it for the code to work with `Masked` instances, since `np.array` will turn those into regular `ndarray`, while `np.stack` will produce a `Masked` instance.

Reply:
Good call! The `np.roll` was for maintaining the column structure -- this gets fixed while dealing with multi-dimensional `Time` objects anyway.